liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -1174,7 +1180,11 @@ private void libertyDependencyWarning(boolean generateFeatures, Exception e) {
         public void installFeatures(File configFile, File serverDir, boolean generateFeatures) {
             try {
                 ServerFeatureUtil servUtil = getServerFeatureUtil(true, libertyDirPropertyFiles);
-                Set<String> features = servUtil.getServerFeatures(serverDir, libertyDirPropertyFiles);
+                FeaturesPlatforms fp = servUtil.getServerFeatures(serverDir, libertyDirPropertyFiles);
+                Set<String> features = null;
+                if (fp != null) {
+                	features = fp.getFeatures();
```

Comment:
Looking through what calls this `installFeatures` method, I think a change in `pom.xml` to point to a different server xml file triggers this scenario. In that case, it seems like both the features and platforms should be considered and passed along to the `runLibertyMojoInstallFeature` further down in this method. The other path through this method is for generateFeatures and does not need to consider platform elements.

Reply:
Actually, this may get handled by the fact the new server xml gets copied to the serverDir. Then the install-feature goal will process the features/platforms from the server.xml file. Need to test this on the cmd line running dev mode.

Reply:
Adding tests today